examples/with-apollo/lib/apolloClient.js

```diff
@@ -43,7 +45,20 @@ export function initializeApollo(initialState = null) {
   return _apolloClient
 }
 
-export function useApollo(initialState) {
-  const store = useMemo(() => initializeApollo(initialState), [initialState])
+export function addApolloState(client, pageProps) {
+  return {
+    ...pageProps,
+    props: {
+      ...pageProps.props,
+      [APOLLO_STATE_PROP_NAME]: client.cache.extract(),
+    },
+  }
```

Comment:
This can be easier to understand and provider the same result, just an idea, you don't have to merge it.

```suggestion
  pageProps.props[APOLLO_STATE_PROP_NAME] = client.cache.extract()
  return pageProps
```